### ISSUE TITLE
javax.xml.bind.UnmarshalException

### ISSUE THREAD

**Issue by ArturLorek** [veraPDF/veraPDF-library#1160]
GUI works quite OK, but the CLI invoked verapdf gets errors right off the bat.
Is there anything I could do to make it work ??

C:\Program Files\VeraPDF>verapdf.bat --off --extract "D:\Verapdf\test file.pdf"
javax.xml.bind.UnmarshalException
 - with linked exception:
[org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.]
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.createUnmarshalException(Unknown Source)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.createUnmarshalException(UnmarshallerImpl.java:548)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:234)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:199)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(Unknown Source)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(Unknown Source)
        at org.verapdf.apps.VeraAppConfigImpl.fromXml(VeraAppConfigImpl.java:271)
        at org.verapdf.apps.ConfigManagerImpl.getApplicationConfig(ConfigManagerImpl.java:154)
        at org.verapdf.cli.VeraPdfCliProcessor.<init>(VeraPdfCliProcessor.java:91)
        at org.verapdf.cli.VeraPdfCliProcessor.createProcessorFromArgs(VeraPdfCliProcessor.java:143)
        at org.verapdf.cli.VeraPdfCli.singleThreadProcess(VeraPdfCli.java:130)
        at org.verapdf.cli.VeraPdfCli.main(VeraPdfCli.java:101)
        at org.verapdf.apps.GreenfieldCliWrapper.main(GreenfieldCliWrapper.java:54)
Caused by: org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 1; Content is not allowed in prolog.
        at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.createSAXParseException(Unknown Source)
        at com.sun.org.apache.xerces.internal.util.ErrorHandlerWrapper.fatalError(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLErrorReporter.reportError(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLScanner.reportFatalError(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:228)
        ... 10 more
Exception in thread "main" java.lang.NullPointerException
        at java.util.EnumSet.typeCheck(Unknown Source)
        at java.util.RegularEnumSet.add(Unknown Source)
        at java.util.RegularEnumSet.add(Unknown Source)
        at com.sun.xml.bind.v2.runtime.reflect.Lister$CollectionLister.addToPack(Lister.java:274)
        at com.sun.xml.bind.v2.runtime.reflect.Lister$CollectionLister.addToPack(Lister.java:238)
        at com.sun.xml.bind.v2.runtime.unmarshaller.Scope.add(Scope.java:91)
        at com.sun.xml.bind.v2.runtime.property.ArrayERProperty$ReceiverImpl.receive(ArrayERProperty.java:183)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.endElement(UnmarshallingContext.java:585)
        at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.endElement(SAXConnector.java:150)
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.endElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
        at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
        at com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:228)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:206)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:261)
        at org.verapdf.core.XmlSerialiser.typeFromXml(XmlSerialiser.java:135)
        at org.verapdf.core.XmlSerialiser.typeFromXml(XmlSerialiser.java:118)
        at org.verapdf.features.FeatureFactory.configFromXml(FeatureFactory.java:63)
        at org.verapdf.apps.ConfigManagerImpl.getFeaturesConfig(ConfigManagerImpl.java:91)
        at org.verapdf.cli.VeraPdfCliProcessor.<init>(VeraPdfCliProcessor.java:93)
        at org.verapdf.cli.VeraPdfCliProcessor.createProcessorFromArgs(VeraPdfCliProcessor.java:143)
        at org.verapdf.cli.VeraPdfCli.singleThreadProcess(VeraPdfCli.java:130)
        at org.verapdf.cli.VeraPdfCli.main(VeraPdfCli.java:101)
        at org.verapdf.apps.GreenfieldCliWrapper.main(GreenfieldCliWrapper.java:54)

**Comment by bdoubrov** [comment]
Would you please specify Java version and OS of your machine? It would also help if you could provide the PDF file you use for tests and the (zipped) contents of your config folder. It looks like the CLI has problems parsing some of the XML configuration files. But we can't yet reproduce this on our side.

**Comment by ArturLorek** [comment]
Hello;
Thank you kindly for your prompt response to this issue.

I am running this on a Windows 10, ver 20H2 build 19042.928.
VeraPDF was installed using a downloaded package and the installer
contained within.
The only difference to a default installation, was a custom location. As
you may have already gathered I installed vera under: C:\Program
Files\VeraPDF instead of a user folder tree.

I am running Java 8 update 171 (build 1.8.0_171-b11) as shown below.

C:\WINDOWS\system32>java -version
java version "1.8.0_171"
Java(TM) SE Runtime Environment (build 1.8.0_171-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.171-b11, mixed mode)

As I already said - it works fine with the GUI, but fails on the CLI.

If I could also ask for not distributing the document anywhere, as it is a
CDA bound document from a Bioscience company and shall not be distributed.
Basically what I am trying to achieve is to use Vera to determine the
security of the PDF in general - whether imposed via the security panel, or
protection with a password, certificate etc.

Look forward to hearing from you on this issue.

KR
Arthur

**Comment by bdoubrov** [comment]
Hi Artur, 

We are still not able to reproduce this behavior. This issue seems to be related with parsing the configuration XML files. However, both CLI and GUI use the same config, and the parsing code is shared between them. 

Could you check that you indeed run the CLI ad GUI from the same veraPDF installation? If the issue persists, please attach here the archive of the config folder in your veraPDF installation.

**Comment by bdoubrov** [comment]
Fix released in v1.20
